### PR TITLE
[lodash] fix _.map to choose array overload signature given an array instead of tuple signature

### PR DESCRIPTION
**Pull request by Ha-limLee**
Since #70840, _.map's overload signature couldn't differentiate between array and tuple. So `_.map(array)` was matched to tuple signature, resulting in the iteratee being `TupleIterator` instead of `ArrayIterator` .

image

This PR fixes the problem. Now it works as intended.

image

See: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70840#issuecomment-2431359820

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://lodash.com/docs/4.17.15#map>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/lodash/common/collection.d.ts [hunk 1]
```diff
@@ -1153,7 +1153,7 @@ declare module "../index" {
          * @param iteratee The function invoked per iteration.
          * @return Returns the new mapped array.
          */
-        map<T extends readonly unknown[], TResult>(collection: T, iteratee: TupleIterator<T, TResult>): { [K in keyof T]: TResult };
+        map<T extends readonly unknown[], TResult>(collection: IsTuple<T>, iteratee: TupleIterator<T, TResult>): { [K in keyof T]: TResult };
```
Comment (aj-r) [inline types/lodash/common/collection.d.ts:1156]: Would this work instead?

`<T extends [unknown, ...readonly unknown[],`

And remove IsTuple?

**Review by aj-r — CHANGES_REQUESTED** [review]

File: types/lodash/common/collection.d.ts (hunk 1 above)
Comment (Ha-limLee) [reply types/lodash/common/collection.d.ts]: Thanks. That's much simpler.

**Review by aj-r — APPROVED** [review]

**Comment by chadlwilson** [comment]
Not familiar with the lifecycle and bot automation here as looks like the approvals are here, but are we able to get this merged? Types are still broken for this "critical package" right now :-/

**Comment by chadlwilson** [comment]
Ahh OK, so needs a second approval.

Of course I've downgraded but fresh installs are broken, that's all. Not to worry.

**Comment by Ha-limLee** [comment]
Ready to merge

**Pull request merged**